patches/minecraft/net/minecraft/block/BlockReed.java.patch

```diff
@@ -13,23 +13,30 @@
      public boolean func_176196_c(World p_176196_1_, BlockPos p_176196_2_)
      {
          Block block = p_176196_1_.func_180495_p(p_176196_2_.func_177977_b()).func_177230_c();
-+        if (block.canSustainPlant(p_176196_1_, p_176196_2_.func_177977_b(), EnumFacing.UP, this)) return true;
++        if (net.minecraftforge.common.SoilType.canPlantStayDefault(p_176196_1_, p_176196_2_, this, p_176196_1_.func_180495_p(p_176196_2_))) return true;
  
          if (block == this)
          {
-@@ -164,4 +165,15 @@
+@@ -164,4 +165,22 @@
      {
          return new BlockState(this, new IProperty[] {field_176355_a});
      }
 +
 +    @Override
-+    public net.minecraftforge.common.EnumPlantType getPlantType(IBlockAccess world, BlockPos pos)
++    public java.util.Set<net.minecraftforge.common.SoilType> getSoilTypes(IBlockAccess world, BlockPos pos, IBlockState state)
 +    {
-+        return net.minecraftforge.common.EnumPlantType.Beach;
++        return java.util.Collections.singleton(net.minecraftforge.common.SoilType.Beach);
```

Comment:
My reading of this is that other beach blocks can be planted on this one. I'm pretty sure that's not intended.